src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java

```diff
@@ -107,7 +107,8 @@ private CloseableHttpClient createHttpClient() throws IOException {
     // basic auth
     CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
     if (!Strings.isNullOrEmpty(config.getUsername()) && !Strings.isNullOrEmpty(config.getPassword())) {
-      AuthScope authScope = new AuthScope(HttpHost.create(config.getUrl()));
+      URI uri = URI.create(config.getUrl());
+      AuthScope authScope = new AuthScope(new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme()));
```

Comment:
Should we use `HttpHost.create(URI uri)` which seems to be doing the same thing internally?

Reply:
I believe there is a confusion with org.apache.commons.httpclient.HttpHost and org.apache.http.HttpHost(used here) class. org.apache.commons.httpclient.HttpHost have the method with URI as argument but org.apache.http.HttpHost does not.